src/irc.py

```diff
@@ -139,42 +175,82 @@ def _write(self, path: pathlib.Path, source: str) -> None:
         path.write_text(source, encoding="utf-8")
         logger.info("Pushed file %s", path)
 
-    def _generate_PEM_file_local(self) -> str:
+    def _generate_PEM_file_local(self) -> None:
         """Generate the PEM file content.
 
-        Returns:
-            A string
+        Raises:
+            
         """
-        pass
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem ]] || "
+            + "openssl genpkey -out {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem "
+            + "-outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048",
```

Comment:
I think some parameters here (like algo and key size) should be constants of the charm